src/appshell/view/welcomedialogmodel.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "welcomedialogmodel.h"
+#include "log.h"
+
+using namespace mu::appshell;
+
+WelcomeDialogModel::WelcomeDialogModel()
+{
+}
+
+void WelcomeDialogModel::init()
+{
+    IF_ASSERT_FAILED(startupScenario() && configuration()) {
+        return;
+    }
+
+    m_items = startupScenario()->welcomeDialogData();
+
+    m_currentIndex = configuration()->welcomeDialogLastShownIndex();
+    nextItem();
+
+    IF_ASSERT_FAILED(m_currentIndex != muse::nidx) {
+        m_currentIndex = 0;
+    }
+    configuration()->setWelcomeDialogLastShownIndex(static_cast<int>(m_currentIndex));
+
+    emit itemsChanged();
+    emit currentItemChanged();
+}
+
+QVariantMap WelcomeDialogModel::currentItem() const
+{
+    if (m_items.empty()) {
+        return QVariantMap();
+    }
+    return m_items.at(m_currentIndex);
+}
+
+void WelcomeDialogModel::nextItem()
+{
+    IF_ASSERT_FAILED(!m_items.empty()) {
+        return;
+    }
+
+    if (hasNext()) {
+        ++m_currentIndex;
+    } else {
+        // Cycle to first...
+        m_currentIndex = 0;
+    }
+    configuration()->setWelcomeDialogLastShownIndex(m_currentIndex);
+
+    emit currentItemChanged();
+}
+
+void WelcomeDialogModel::prevItem()
+{
+    IF_ASSERT_FAILED(!m_items.empty()) {
+        return;
+    }
+
+    if (hasPrev()) {
+        --m_currentIndex;
+    } else {
+        // Cycle to last....
+        m_currentIndex = count() - 1;
+    }
+    configuration()->setWelcomeDialogLastShownIndex(m_currentIndex);
```

Comment:
please fix this
`'argument': conversion from 'size_t' to 'int', possible loss of data`